main_test.go

```diff
@@ -588,6 +642,14 @@ func startHTTP() (net.Listener, chan struct{}) {
 	return ln, done
 }
 
+func startRedis(t *testing.T) *miniredis.Miniredis {
+	redis := miniredis.NewMiniRedis()
+	if err := redis.StartAddr("127.0.0.1:6379"); err != nil {
```

Comment:
It could be better to start redis on a random int port.
```go
rand.Seed(time.Now().UnixNano())
min := 10
max := 30
port := rand.Intn(max - min + 1) + min
```

and to pass that port (or worst case top scope variable) to the underlying tests.